test/integ/test_put_get_compress_enc.py

```diff
@@ -0,0 +1,228 @@
+                )
+
+            ret = cnx.cursor().execute(f"LS @~/{_test_name}").fetchone()
+            uploaded_gz_name = f"{file_name}.gz"
+            if auto_compress:
+                assert uploaded_gz_name in ret[0]
+            else:
+                assert uploaded_gz_name not in ret[0]
+
+            # get this file, if the client handle compression meta correctly
+            get_dir = tmp_path / "get_dir"
+            get_dir.mkdir()
+            cnx.cursor().execute(
+                f"GET @~/{_test_name}/{file_name} file://{get_dir}"
+            ).fetchone()
+            # TODO: The downloaded file should always be the unzip (original) file
+            downloaded_file = get_dir / (
+                uploaded_gz_name if auto_compress else file_name
+            )
+            assert downloaded_file.exists()
+            if not auto_compress:
+                assert filecmp.cmp(test_data, downloaded_file)
+
+        finally:
+            cnx.cursor().execute(f"RM @~/{_test_name}")
+            if file_stream:
+                file_stream.close()
+
+
+@pytest.mark.skipif(hack_s3_rest_client, reason="s3_rest_client could not be imported.")
+def test_get_gzip_content_encoding(
+    tmp_path: pathlib.Path,
+    conn_cnx,
+):
+    """Tests GET command for a content-encoding=GZIP in stage"""
+    _test_name = "test_get_gzip_content_encoding"
+    test_data, file_name = _prepare_tmp_file(tmp_path)
+
+    with patch(
+        "snowflake.connector.s3_storage_client.SnowflakeS3RestClient._send_request_with_authentication_and_retry",
+        mock_send_request,
+    ):
+        with conn_cnx() as cnx:
+            cnx.cursor().execute(f"RM @~/{_test_name}")
+            try:
+                file_stream = test_data.open("rb")
+                with cnx.cursor() as cur:
+                    put(
+                        cur,
+                        str(test_data),
+                        f"~/{_test_name}",
+                        False,
+                        sql_options="auto_compress=True",
+                        file_stream=file_stream,
+                    ).fetchone()
+
+                ret = cnx.cursor().execute(f"LS @~/{_test_name}").fetchone()
+                assert f"{file_name}.gz" in ret[0]
+
+                # get this file, if the client handle compression meta correctly
+                get_dir = tmp_path / "get_dir"
+                get_dir.mkdir()
+                ret = (
+                    cnx.cursor()
+                    .execute(f"GET @~/{_test_name}/{file_name} file://{get_dir}")
+                    .fetchone()
+                )
+                downloaded_file = get_dir / ret[0]
+                assert downloaded_file.exists()
+
+            finally:
+                cnx.cursor().execute(f"RM @~/{_test_name}")
+                if file_stream:
+                    file_stream.close()
+
+
+@pytest.mark.skipif(hack_s3_rest_client, reason="s3_rest_client could not be imported.")
```

Comment:
Since you marked the whole module as `skipolddriver` I don't think this is necessary

Reply:
mark tests separately.